### PR TITLE
feat(security): apply chmod 600 to audit.log on creation (#531)

### LINKED ISSUE #531: feat(security): apply chmod 600 to audit.log on creation
## Hardening

The security audit log at `~/.egc/audit.log` is written via `appendFileSync` with no explicit permissions. On Linux/macOS, this creates the file with the process umask (typically `644`), making it **readable by all local users**.

This log contains sensitive security events: blocked commands, denied file writes, rate-limit hits, and mutated payloads. Exposing it to other local users undermines the audit trail's confidentiality.

The state files fixed in PR #528 received `chmod 600` — the audit log was not included.

## Proposal

After the first write to `~/.egc/audit.log`, apply `fs.chmodSync(auditLogPath, 0o600)`. Alternatively, open the file with `fs.openSync(auditLogPath, 'a', 0o600)` so it is created with the correct mode from the start.

## Where

`mcp/servers/egc-guardian/src/index.ts` — `writeSecurityAuditLog` function (around line 112).

Good first issue: one-line fix, mirroring the pattern already used for state files.

### PR DESCRIPTION
**Pull request by Maqbool61**
audit.log was written via appendFileSync with no explicit mode, so new files inherited the process umask (typically 644) — readable by all local users on the same machine. The log contains sensitive security events: blocked commands, denied writes, rate-limit hits.

Also closes a second gap: appendFileSync alone would silently drop every DENIED event if ~/.egc didn't exist yet (e.g. fresh install, no prior state-file write).

Fix: create ~/.egc with mode 0700 if missing, write audit.log with mode 0600 at creation time, and chmodSync as a backstop for the case where the file already existed before this fix shipped.

Verified locally: file created at mode 600 on first write, content correct.

Fixes #531

## Testing Done

- [x] Manual testing completed
- [ ] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [x] `fix:` Bug fix
- [ ] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [ ] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [ ] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [ ] No sensitive data exposed in logs or output
- [ ] Follows conventional commits format

## Documentation
- [ ] Updated relevant documentation
- [ ] Added comments for complex logic
- [ ] README updated (if needed)

---
## Summary by cubic
Locks down `audit.log` to 0600 and `~/.egc` to 0700, ensures first-run audit logging, and surfaces write failures via the system logger. Also isolates dir permission fixes from the write path so DENIED events are never lost (fixes #531).

- **Bug Fixes**
  - Ensure `~/.egc` exists and is 0700; force 0700 on pre-existing dirs, with chmod backstop moved outside the write path (best-effort).
  - Create `audit.log` with 0600; backstop existing files with `chmodSync`.
  - Log audit write failures via `sysLogger` as non-fatal ERRORs instead of swallowing errors.

<sup>Written for commit bd68d5b161210b8012afb3df6dc68ef5878f660e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Fmarzochi/EGC/pull/534?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Strengthened security audit logging by writing logs to a dedicated hidden application directory with tighter permissions.
  * Added safer handling to ensure the log directory exists and audit files are created/updated with restricted access.
  * If audit log writing fails, it is reported but does not interrupt normal app behavior.

### CODE REVIEW

File: mcp/servers/egc-guardian/src/index.ts [hunk 1]
```diff
@@ -106,10 +106,15 @@ class PersistentLogger {
...
-const auditLogPath = path.join(os.homedir(), '.egc', 'audit.log');
+const auditLogDir = path.join(os.homedir(), '.egc');
+const auditLogPath = path.join(auditLogDir, 'audit.log');
 function writeSecurityAuditLog(action: string, details: Record<string, unknown>) {
   const entry = JSON.stringify({ timestamp: new Date().toISOString(), action, ...details });
-  try { fs.appendFileSync(auditLogPath, entry + '\n', 'utf-8'); } catch { /* non-critical */ }
+  try {
+    fs.mkdirSync(auditLogDir, { recursive: true, mode: 0o700 });
+    fs.appendFileSync(auditLogPath, entry + '\n', { encoding: 'utf-8', mode: 0o600 });
+    fs.chmodSync(auditLogPath, 0o600);
```
[the remark this answers is not in the document]
Comment (Maqbool61) [reply mcp/servers/egc-guardian/src/index.ts:126]: Fixed in 2949fd5 — added chmodSync(auditLogDir, 0o700) as a backstop, and write failures now surface via sysLogger.log instead of being silently swallowed.

File: mcp/servers/egc-guardian/src/index.ts [hunk 2]
```diff
@@ -106,10 +106,15 @@ class PersistentLogger {
...
 
-const auditLogPath = path.join(os.homedir(), '.egc', 'audit.log');
+const auditLogDir = path.join(os.homedir(), '.egc');
+const auditLogPath = path.join(auditLogDir, 'audit.log');
 function writeSecurityAuditLog(action: string, details: Record<string, unknown>) {
   const entry = JSON.stringify({ timestamp: new Date().toISOString(), action, ...details });
-  try { fs.appendFileSync(auditLogPath, entry + '\n', 'utf-8'); } catch { /* non-critical */ }
+  try {
+    fs.mkdirSync(auditLogDir, { recursive: true, mode: 0o700 });
+    fs.appendFileSync(auditLogPath, entry + '\n', { encoding: 'utf-8', mode: 0o600 });
```
[the remark this answers is not in the document]
Comment (Maqbool61) [reply mcp/servers/egc-guardian/src/index.ts:125]: Fixed in 2949fd5 — added chmodSync(auditLogDir, 0o700) as a backstop, and write failures now surface via sysLogger.log instead of being silently swallowed.

File: mcp/servers/egc-guardian/src/index.ts [hunk 3]
```diff
@@ -106,10 +106,18 @@ class PersistentLogger {
...
 
-const auditLogPath = path.join(os.homedir(), '.egc', 'audit.log');
+const auditLogDir = path.join(os.homedir(), '.egc');
+const auditLogPath = path.join(auditLogDir, 'audit.log');
 function writeSecurityAuditLog(action: string, details: Record<string, unknown>) {
   const entry = JSON.stringify({ timestamp: new Date().toISOString(), action, ...details });
-  try { fs.appendFileSync(auditLogPath, entry + '\n', 'utf-8'); } catch { /* non-critical */ }
+  try {
+    fs.mkdirSync(auditLogDir, { recursive: true, mode: 0o700 });
+    fs.chmodSync(auditLogDir, 0o700);
```
[the remark this answers is not in the document]
Comment (Maqbool61) [reply mcp/servers/egc-guardian/src/index.ts]: Fixed in bd68d5b — moved the directory chmodSync into its own try/catch outside the write's critical path. Verified with a test that simulates chmodSync throwing EPERM on the directory: the DENIED event still gets written to disk correctly.

**Comment by Fmarzochi** [comment]
Thanks @Maqbool61 -- the fix is correct. The `mkdirSync` + `appendFileSync({mode:0o600})` + `chmodSync` backstop pattern covers both the first-run case and pre-existing installs cleanly. Merging now.

For future PRs: please sign your commits with `git commit -s`. It adds a `Signed-off-by:` line required by our Developer Certificate of Origin policy. The CI check is currently informational (non-blocking), but the project asks for it on all contributions.
